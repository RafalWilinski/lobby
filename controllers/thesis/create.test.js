const { omit } = require('lodash');
const create = require("./create");

describe("POST /thesis", () => {
  it("should return 200 if correct thesis is provided", async () => {
    const ctx = {
      request: {
        body: {
          thesis: {
            name: "Testowy Temat",
            description: "Opis Opis Opis Opis Opis Opis Opis Opis Opis Opis ",
            ispublic: false,
            branches: [
              "Przetwarzanie Rozproszone",
              "Analiza Danych",
              "Programowanie Równoległe",
              "Badania Operacyjne"
            ],
            promoterId: 1
          },
          roles: [
            {
              name: "Ja jako kapitan",
              description:
                "Stanowisko Stanowisko Stanowisko Stanowisko Stanowisko Stanowisko Stanowisko ",
              skills: ["C", "C++", "C#", "Python"],
              capitan: true,
              userLogin: "1@1.com"
            },
            {
              name: "Druga pozycja do testowego",
              description: "Wymagane skille: Arduino, SQL, NoSQL",
              skills: ["Arduino", "SQL", "NoSQL"],
              capitan: false
            }
          ]
        }
      }
    };

    await create(ctx);

    expect(ctx.body.thesis.description).toMatchSnapshot();
  });
});
