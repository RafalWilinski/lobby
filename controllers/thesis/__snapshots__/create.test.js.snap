// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /thesis should return 200 if correct thesis is provided 1`] = `"Opis Opis Opis Opis Opis Opis Opis Opis Opis Opis "`;
